hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMDirectoriesPurgeRequestWithFSO.java

```diff
@@ -120,7 +120,11 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager, TermIn
             long parentId = keyInfo.getParentObjectID();
             dbOpenKey = omMetadataManager.getOpenFileName(path.getVolumeId(), path.getBucketId(),
                 parentId, keyInfo.getFileName(), hsyncClientId);
-            openKeyRemovalList.add(dbOpenKey);
+            OmKeyInfo openKeyInfo = omMetadataManager.getOpenKeyTable(getBucketLayout()).get(dbOpenKey);
+            if (openKeyInfo != null) {
+              openKeyInfo.getMetadata().put(DELETED_HSYNC_KEY, "true");
```

Comment:
But what happens when the same client tries to create a key at the same path?

Reply:
Tell me if this would happen:

- When the same client (with the same client ID, thus same `dbOpenKey`) is allowed to write to the same path, the `openKeyTable` (`openFileTable`) entry will be overwritten, still leading to orphan blocks.

On the other hand, it also doesn't make sense to block the same client from writing to the same key path, just because the key is deleted by another client or because the key is errorneously deleted without being close()d first.

Reply:
I think this problem exist even for non `hsync` keys as well.
When Client1 is writing a key, and some other client comes and deletes the directory or bucket and recreate again. Client1 will fail to commit the key and also will not able to write same key till `openKey` is cleaned up or it may overwrite.

Reply:
> will not able to write same key till `openKey` is cleaned up

Got it. So at least in this edge case it won't cause new orphan blocks. But we should revisit if this is the desired behavior.

Do we have CLI command to manually trigger a run of `OpenKeyCleanup`?

Reply:
I checked but don't find any CLI to manually trigger `OpenKeyCleanup.`